test/SSTFrequencyLimitDetectorTest.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "MockPlatformIO.hpp"
+#include "MockPlatformTopo.hpp"
+#include "SSTFrequencyLimitDetector.hpp"
+#include "geopm/Helper.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm_test.hpp"
+
+using ::testing::_;
+using testing::AllOf;
+using testing::ElementsAre;
+using testing::Ge;
+using ::testing::Invoke;
+using testing::Lt;
+using testing::Pair;
+using ::testing::Return;
+
+using geopm::SSTFrequencyLimitDetector;
+
+static const double CPU_FREQUENCY_MAX = 3.7e9;
+static const double ALL_CORE_TURBO_LIMIT = 2.7e9;
+static const double CPU_FREQUENCY_STICKER = 2.1e9;
+static const double CPU_FREQUENCY_STEP = 1e8;
+static const double LP_FREQ_SSE = 2.4e9;
+static const double LP_FREQ_AVX2 = 2.1e9;
+static const double LP_FREQ_AVX512 = 1.7e9;
+
+static const int CORE_COUNT = 4;
+static const std::vector<double> HP_CORES = {2, 3, 4};
+static const std::vector<double> HP_FREQS_SSE = {3.6e9, 3.3e9, 3.0e9};
+static const std::vector<double> HP_FREQS_AVX2 = {3.5e9, 3.2e9, 2.9e9};
+static const std::vector<double> HP_FREQS_AVX512 = {3.4e9, 3.1e9, 2.8e9};
+
+static const int CLOS_SIGNAL_INDEX_OFFSET = 100;
+static const int SST_ENABLE_SIGNAL_INDEX_OFFSET = 1000;
```

Comment:
Not important, but these could go in the test fixture class (just to have everything related to the tests neatly encapsulated).